### PR TITLE
Deploy maven site to gwt-maven-plugin.github.com

### PR DESCRIPTION
**Pull request by ibaca**
Requires CI_SITE_USERNAME and CI_SITE_PASSWORD (github account) to be defined in .travis.yml.

### CODE REVIEW

**Comment by tbroyer** [comment]
IMO the site should only be deployed during releases.
For tbroyer/gwt-maven-plugin I'm using the standard site plugin with a custom gh-pages "wagon" and it seems to work (site-deploy correctly deploys the site).

Re. gwt-maven-plugin.github.io vs mojohaus, I think someone should ask mojohaus first. Also, I think they setup an automatic redirect from codehaus to mojohaus, so using gh-pages here would lessen the visibility of the project.

**Comment by ibaca** [comment]
I thought that we can deploy snapshots to gwt-maven-plugin.github.io and release to mojohaus. The configuration is quite simple (difficult to say what's better/simpler, the com.github.github:site-maven-plugin vs gitsite wagon) and I don't think travis will complain for the extra work. Anyways, I'll try create a pull request to 'deploy to mojohouse'.

**Comment by tbroyer** [comment]
> I thought that we can deploy snapshots to gwt-maven-plugin.github.io and release to mojohaus.

Ah yeah, why not.

I'll add you as a commiter and if you don't get any other feedback then you can go ahead (and configure your credentials in Travis I suppose ;-) )

**Comment by tbroyer** [comment]
Cc @niloc132

**Comment by ibaca** [comment]
I've updated the way to deploy using gitsite wagon, so the site:deploy still deploy to codehaus and site:stage-deploy points to the snapshot gwt-maven-plugin.github.io.

I think that mojohaus is still migrating its plugins and updating the parent-pom and releases procedures. So I suppose that the site:deploy cannot be fixed until mojohaus endup the migration. In the mean time users can use staging site.

**Comment by tbroyer** [comment]
I think it's enough to push the site to the mojohaus/gwt-maven-plugin's repo gh-pages' branch to get it at www.mojohaus.org. Maybe use that repo only for the online docs for releases?

**Comment by ibaca** [comment]
It's true, mojohoaus team finally describe how to upgrade each plugin site http://blog.soebes.de/blog/2015/06/12/making-github-page-for-mojohaus/ (create maven site manually an push to gh-pages. I'll update this pull request to apply the release publish too. But someone with access to mojohaus/gwt-maven-plugin can publish the site now manually :wink:.

I was having problems with authentication, current pull request state was not working using public key in travis. I solved using this strategy (http://grosser.it/2014/03/01/allowing-travis-to-git-clone-by-adding-a-public-ssh-key/ add private key using script). But! I'll try to add the private key as an encrypted var to the travis.yml and it fails because is too large. I only get work using environment variable configured manually using the travis repo settings page. This is enough, isn't it? (all this is less important if the mojohaus site get published).

**Comment by tbroyer** [comment]
> But someone with access to mojohaus/gwt-maven-plugin can publish the site now manually .

FWIW, I don't have such access rights.

> I was having problems with authentication, current pull request state was not working using public key in travis.

How about using HTTPS instead of SSH? Using a “personal access token”.

**Comment by ibaca** [comment]
Yes, this is an alternative, but currently gitsite only support git:// or file:// protocols. I modify the source code to allow https, but is not that easy :disappointed:. I was really close to get it work using public keys... I'll try again later. If I don't get it work, I'll rollback to my initial strategy using the github site plugin which support https.

**Comment by ibaca** [comment]
This last version is working, I added the private key in travis and the pub key here in github.

The problems was very suspicious; not sure, but or some extremely dummy error or a critical security problem was happening. As you can see here https://travis-ci.org/ibaca/rxjava-gwt/builds/67098609#L1322 github was thinking that I am 'some-public-token', which looks like a ssl/github indicating a random public token, but, after some research, the name after "Hi XXX!" is the actual user name. This user https://github.com/some-public-token. Not sure, very strange. This doesn't happen anymore.

File: generated-archetype/some-artifact/src/test/java/com/company/client/GwtTestSomeModule.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
  * 
  * If you run the tests using the Maven command line, you will have to 
  * navigate with your browser to a specific url given by Maven. 
- * See http://mojo.codehaus.org/gwt-maven-plugin/user-guide/testing.html 
+ * See http://gwt-maven-plugin.github.com/gwt-maven-plugin/user-guide/testing.html
```
Comment (tbroyer) [inline generated-archetype/some-artifact/src/test/java/com/company/client/GwtTestSomeModule.java]: Use https and github.io

File: src/it/gwt-test-fail/src/test/java/fr/salvadordiaz/gwt/client/FailTestGwt.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 /**
  * Failing GwtTestCase <br/>
- * read : http://mojo.codehaus.org/gwt-maven-plugin/user-guide/testing.html pour plus d'informations
+ * read : http://gwt-maven-plugin.github.com/gwt-maven-plugin/user-guide/testing.html pour plus d'informations
```
Comment (tbroyer) [inline src/it/gwt-test-fail/src/test/java/fr/salvadordiaz/gwt/client/FailTestGwt.java]: `https://gwt-maven-plugin.github.io/…`

File: src/site/apt/eclipse/google_plugin.apt.vm [hunk 3]
```diff
@@ -144,7 +144,7 @@ Using the Google Eclipse Plugin
   configured (*) as a GWT module project, changes to java sources will be available in DevMode with a simple
   refresh with no requirement to repackage the modules. 
   
-[../images/eclipse/gwt-maven-logo.png] m2eclipse detecting the gwt module as project reference.
+[../images/gwt-maven-logo.png] m2eclipse detecting the gwt module as project reference.
```
Comment (tbroyer) [inline src/site/apt/eclipse/google_plugin.apt.vm]: Now that both GWT and Maven have new logos, I think that one needs to be updated ;-)

**Comment by tbroyer** [comment]
SGTM (not that I really care about it…)

**Comment by ibaca** [comment]
good enough?

**Comment by ibaca** [comment]
I just created the github page (which is required before travis build) and merged the pull request. I hope is ok.

**Comment by ibaca** [comment]
Both scripts are failing, I'm trying to solve. 

deploy-snapshot.sh
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy (default-deploy) on project gwt-maven-plugin: Failed to deploy artifacts: 
    Could not transfer artifact org.codehaus.mojo:gwt-maven-plugin:jar:2.8.0-20150628.105129-9 from/to codehaus-nexus-snapshots (https://nexus.codehaus.org/content/repositories/snapshots/): 
    Failed to transfer file: https://nexus.codehaus.org/content/repositories/snapshots/org/codehaus/mojo/gwt-maven-plugin/2.8.0-SNAPSHOT/gwt-maven-plugin-2.8.0-20150628.105129-9.jar. 
    Return code is: 401, ReasonPhrase: Unauthorized. -> [Help 1]
```
deploy-site.sh
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.4:site (default-site) on project gwt-maven-plugin: 
    Execution default-site of goal org.apache.maven.plugins:maven-site-plugin:3.4:site failed: org.apache.maven.reporting.MavenReportException: 
    Could not find resource 'http://svn.codehaus.org/mojo/tags/mojo-parent-34/src/main/config/pmd/mojo_rules.xml'. -> [Help 1]
```

I re-started the last successfully completed job (https://travis-ci.org/gwt-maven-plugin/gwt-maven-plugin/jobs/63955726#L372) as a simple check if the problem has not been introduced in this pull request, and the 'deploy-snapshot 401 Unauthorized' is returned too. This is probably something related to mojohaus updating its services (not confirmed).

**Comment by tbroyer** [comment]
My credentials indeed no longer work on nexus.codehaus.org. Please contact the MojoHaus team and ask for access, and then you could update the Travis CI configuration.

**Comment by ibaca** [comment]
I had deployed the site manually and I'm waiting for mojohaus (or sonatype) news about access problems (https://groups.google.com/forum/#!topic/mojohaus-dev/70g4q3EdTXg).

**Comment by ibaca** [comment]
FYI @tbroyer the thread to request access to nexus https://groups.google.com/forum/#!topic/mojohaus-dev/-3DXyqeojOQ.

**Pull request merged**